packages/flutter/lib/src/material/text_field.dart

```diff
@@ -365,31 +371,7 @@ class TextField extends StatefulWidget {
        assert(clipBehavior != null),
        assert(enableIMEPersonalizedLearning != null),
        keyboardType = keyboardType ?? (maxLines == 1 ? TextInputType.text : TextInputType.multiline),
-       enableInteractiveSelection = enableInteractiveSelection ?? (!readOnly || !obscureText),
-       toolbarOptions = toolbarOptions ??
```

Comment:
I've moved this logic here: https://github.com/flutter/flutter/pull/107193/files#diff-b4ecc433d218176fa9f2815f6b88ce0eb901452d07aa353e6cf44e8e6f4aa5d8R106

ToolbarOptions should continue to work as it did before.  I removed this default here so that I can distinguish between a user that didn't pass anything here and someone that is explicitly using toolbarOptions.